rotkehlchen/inquirer.py

```diff
@@ -766,136 +839,185 @@ def find_usd_price_and_oracle(
+        prices, assets_without_price = Inquirer._try_oracle_price_query(  # type: ignore[assignment]  # Manual current oracle uses Asset type
             oracle=CurrentPriceOracle.MANUALCURRENT,
             oracle_instance=Inquirer._manualcurrent,
-            from_asset=asset,  # type: ignore[arg-type]  # Manual current oracle can works with Asset type
+            from_assets=assets_without_price,  # type: ignore[arg-type]  # Manual current oracle works with Asset type
             to_asset=A_USD.resolve_to_asset_with_oracles(),
-            coming_from_latest_price=coming_from_latest_price,
-        )) != (ZERO_PRICE, False):
-            price, _ = price_result
-            return price, CurrentPriceOracle.MANUALCURRENT
-
-        # Try and check if it is an ethereum token with specified protocol or underlying tokens
-        is_known_protocol = False
-        underlying_tokens = None
-        if isinstance(asset, EvmToken):
-            if asset.protocol is not None:
-                is_known_protocol = asset.protocol in ProtocolsWithPriceLogic
-            underlying_tokens = asset.underlying_tokens
-
-            # Check if it is a special token
-            if asset.identifier in Inquirer.special_tokens:
-                ethereum = Inquirer.get_evm_manager(chain_id=ChainID.ETHEREUM)
-                underlying_asset_price, oracle = get_underlying_asset_price(asset)
-                usd_price = handle_defi_price_query(
-                    ethereum=ethereum.node_inquirer,  # type:ignore  # ethereum is an EthereumManager so the inquirer is of the expected type
-                    token=asset,
-                    underlying_asset_price=underlying_asset_price,
-                )
-                price = ZERO_PRICE if usd_price is None else Price(usd_price)
-
-                Inquirer.set_cached_price(
-                    cache_key=cache_key,
-                    cached_price=CachedPriceEntry(
-                        price=price,
-                        time=ts_now(),
-                        oracle=CurrentPriceOracle.BLOCKCHAIN,
-                    ),
-                )
-                return price, oracle
-
-            if is_known_protocol is True or underlying_tokens is not None:
-                if (
-                    underlying_tokens is not None and
-                    asset.evm_address in (x.address for x in underlying_tokens)
-                ):
-                    Inquirer._msg_aggregator.add_error(
-                        f'Token {asset} has itself as underlying token. Please edit the '
-                        'asset to fix it. Price queries will not work until this is done.',
+        )
+        for asset, price in prices.items():
+            found_prices[asset] = price, CurrentPriceOracle.MANUALCURRENT
+
+        for asset in assets_without_price:
+            # Try and check if it is an ethereum token with specified protocol or underlying tokens
+            is_known_protocol = False
```

Comment:
things like these should be in another subfunction to be called in the for loop.  The code in this PR has become unreadable. 

The inquirer code was already hard to read and needed improvement. This PR makes it a lot worse.